test/framework/toolchain.py

```diff
@@ -303,16 +306,18 @@ def test_override_optarch(self):
                 flag = None
                 if optarch_var is not None:
                     flag = '-%s' % optarch_var
-                else:
-                    # default optarch flag
+
+                # default optarch flag if defined
+                elif tc.arch in tc.COMPILER_OPTIMAL_ARCHITECTURE_OPTION:
```

Comment:
same here, make sure the major architectures have a matching `optarch` option, that should never disappear